### ISSUE TITLE
kivy settings frezzes GUI to be updated

### ISSUE THREAD

**Issue by jeff-dh** [kivy/kivy#8047]
**Software Versions**
* Python: Python 3.8.10
* OS: Linux Mate
          Linux *** 5.4.0-131-generic # 147-Ubuntu SMP Fri Oct 14 17:07:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
* Kivy: default pip wheel (v2.1.0, git_hash: 023bd79b90f9831b45bb8eb449346648aa5fe5f8) & head of master (90400b656e3749eb6f9cb912f224146f0b06082e)

* Kivy `pip install "kivy @ git+https://github.com/kivy/kivy.git@90400b656e3749eb6f9cb912f224146f0b06082e"` in venv

**Describe the bug**
https://stackoverflow.com/questions/73099115/python-kivy-settings-frezzes-gui-to-be-updated

I had the same issue and it still exists and looks to me like a bug.

The issue is:

The GUI (or at least parts of it) "freeze" after the Settings Panel has been opened and closed. `Widget.text` properties are not updated anymore after the settings panel has been opened and closed.

As mentioned in the stackoverflow post this fixes the issue:

`from kivy.uix.settings import SettingsWithSpinner`

But if the import is missing, no error is reported or similar.

**Expected behavior**
Either the import should not be necessary or I guess the missing import should raise an error.

**To Reproduce**

```python
from kivy.app import App
from kivy.uix.boxlayout import BoxLayout

#uncomment the following line to fix the issue
#from kivy.uix.settings import SettingsWithSpinner

from kivy.lang import Builder

Builder.load_string('''
<Interface>:
    Button:
        text: txb.text
        on_release: app.open_settings()

    TextInput:
        id: txb
        text: "aaa"
''')

class Interface(BoxLayout):
    pass

class InterfaceApp(App):
    def build(self):
        return Interface()


InterfaceApp().run()

```

**Comment by Julian-O** [comment]
FYI: Not reproducible on Windows 11, Kivy 2.2.1

**Comment by misl6** [comment]
Should have been fixed via #8535, please let us know if the issue persists on your side.
